pkg/kvstore/trace.go

```diff
@@ -42,7 +44,7 @@ func Trace(format string, err error, fields logrus.Fields, a ...interface{}) {
 
 func init() {
 	if strings.ToLower(Debug) == "true" {
-		log.SetLevel(logrus.DebugLevel)
+		logging.DefaultLogger.SetLevel(logrus.DebugLevel)
```

Comment:
Is this really necessary? I thought @nebril made a fix to configure logger before any package uses it

Reply:
It's for unit tests. This change is only needed because `log` is no longer equal to the DefaultLogger but instead a `logrus.Entry` because of a `WithFields()`.

Reply:
debugdetection package checks for debug mode in runtime, this Debug var is set at compile time, so it should be fine as long as we don't build production cilium with `-ldflags "-X "github.com/cilium/cilium/pkg/kvstore".Debug=true"`